File: tutorial/examples/core_components/button.py
import dash
import dash_html_components as html
import dash_core_components as dcc

app = dash.Dash()
app.layout = html.Div([
    html.Div(dcc.Input(id='input-box', type="text")),
    html.Button('Submit', id='button'),
    html.Div(id='output-container-button',
             children="Enter a value and press submit")
])

@app.callback(
    dash.dependencies.Output('output-container-button', 'children'),
    [dash.dependencies.Input('button', 'n_clicks')],
    [dash.dependencies.State('input-box', 'value')])
def update_output(n_clicks, value):
    return "The input value was \"{}\" and the button has been clicked {} times".format(
        value,
        n_clicks
    )

if __name__ == '__main__':
    app.run_server(debug=True)

File: dashr/chapters/dash-core-components/button/examples/button.R
library(dashR)
library(dashCoreComponents)
library(dashHtmlComponents)

app <- Dash$new()

app$layout(
  htmlDiv(
    list(
      htmlDiv(dccInput(id="input-box", type="text")),
      htmlButton("Submit", id="button"),
      htmlDiv(id="output-container-button",
              children="Enter a value and press submit")
    )
  )
)

app$callback(
  output(id = "output-container-button", property = 'children'),
  params=list(input(id = "button", property = "n_clicks"),
              input(id = "input-box", property = "value")),
  function(n_clicks, value) {
    sprintf("The input value was \'%s\' and the button has been clicked \'%s\' times", value, n_clicks)
  })

app$run_server()
